addon/components/truncate-multiline.js

```diff
@@ -237,3 +237,32 @@ export default Ember.Component.extend(ResizeHandlerMixin, {
     }
   }
 });
+
+/**
+ * Helper function to determine if an attribute has changed.
+ * @private
+ * @param attrs {Object}
+ * @param name {String}
+ * @return {Boolean}
+ */
+function didAttrChange(attrs, name) {
+  let oldAttr = attrs.oldAttrs[name];
+  let newAttr = attrs.newAttrs[name];
+
+  if (typeof oldAttr === "undefined" && typeof newAttr === "undefined") {
+    return false;
+  }
+
+  /**
+  * In integration tests, the attributes passed to 'didUpdateAttrs' are objects,
+  * while in the app they are the values of the attributes themselves.
+  *
+  * However, if the value of the passed attribute hasn't changed it is NOT an object,
+  * it is the value itself.
+  */
+  oldAttr = oldAttr && oldAttr.value || oldAttr || {};
+  newAttr = newAttr && newAttr.value || newAttr || {};
```

Comment:
This is more complex than it needs to be because the attributes can be undefined. Let's handle all undefined cases above – adding a subsequent OR case that returns true should suffice – and reduce the complexity here.